src/bentoml/_internal/cloud/deployment.py

```diff
@@ -79,28 +127,161 @@ def get_real_bento_tag(
 
 
 @attr.define
-class DeploymentInfo:
-    __omit_if_default__ = True
-    name: str
+class DeploymentConfig(CreateDeploymentSchemaV2):
+    def to_yaml(self):
+        return yaml.dump(bentoml_cattr.unstructure(self))
+
+    def to_dict(self):
+        return bentoml_cattr.unstructure(self)
+
+
+@attr.define
+class DeploymentState:
+    status: str
     created_at: str
-    bento: Tag
-    status: DeploymentStatus
-    admin_console: str
-    endpoint: t.Optional[str]
-    config: dict[str, t.Any]
+    updated_at: str
+    # no error message for now
+    # error_msg: str
 
-    def to_dict(self) -> t.Dict[str, t.Any]:
+    def to_dict(self) -> dict[str, t.Any]:
         return bentoml_cattr.unstructure(self)
 
 
 @attr.define
-class Deployment:
-    context: t.Optional[str]
-    cluster_name: str
+class DeploymentInfo:
     name: str
+    admin_console: str
+    created_at: str
+    created_by: str
+    cluster: str
+    organization: str
+    distributed: bool
+    description: t.Optional[str]
+    _context: t.Optional[str] = attr.field(alias="_context", repr=False)
     _schema: DeploymentSchema = attr.field(alias="_schema", repr=False)
-    _urls: t.Optional[list[str]] = attr.field(alias="_urls", default=None)
+    _urls: t.Optional[list[str]] = attr.field(alias="_urls", default=None, repr=False)
+
+    def to_dict(self) -> dict[str, t.Any]:
+        return {
+            "name": self.name,
+            "cluster": self.cluster,
+            "description": self.description,
+            "organization": self.organization,
+            "admin_console": self.admin_console,
+            "created_at": self.created_at,
+            "created_by": self.created_by,
+            "distributed": self.distributed,
+            "config": self.get_config(refetch=False).to_dict(),
+            "status": self.get_status(refetch=False).to_dict(),
+        }
+
+    def _refetch(self) -> None:
+        res = Deployment.get(self.name, self.cluster, self._context)
+        attr.evolve(self, _schema=res._schema, _urls=res._urls)
```

Comment:
```suggestion
        self._schema = res._schema
        self._urls = res._urls
```

Let's keep it for now, `attr.evolve` doesn't update the first argument in-place.

Reply:
attr.evolve create a new object.

but never mind since we don't want to make the deployment object read-only.